include/Machine.h

```diff
@@ -6,34 +6,38 @@
 
 #include "FabMember.h"
 
-class Machine
+enum class MachineType
+{
+  INVALID,
+  PRINTER3D,
+  LASER,
+  CNC,
+  EXTRA1,
+  EXTRA2
+};
+
+struct MachineConfig
 {
+  uint16_t machine_id;
+  MachineType machine_type;
+  uint8_t control_pin;
+  bool control_pin_active_low;
 public:
-  typedef uint32_t MachineIDType;
-  typedef enum enum_MachineType
-  {
-    INVALID = 0x0,
-    PRINTER3D = 0x1,
-    LASER = 0x2,
-    CNC = 0x3,
-    EXTRA1 = 0xFE,
-    EXTRA2 = 0xFF
-  } MachineType;
+  MachineConfig() : machine_id(0), machine_type(MachineType::INVALID), control_pin(0), control_pin_active_low(false) {};
+  MachineConfig(uint16_t id, MachineType type, uint8_t pin, bool active_low): machine_id(id), machine_type(type), control_pin(pin), control_pin_active_low(active_low) {}
+};
 
+class Machine
+{
 private:
-  MachineIDType _machine_id = INVALID;
-  MachineType _machine_type = INVALID;
-  uint8_t _control_pin = -1;
-  bool _control_pin_active_low = false;
-
-  bool _active;
-  FabMember _current_user;
-  void _turnOn();
-  void _turnOff();
-  uint32_t _usage_start_timestamp;
+  MachineConfig config;
+  bool active;
+  FabMember current_user;
+  void power(bool on_or_off);
+  uint32_t usage_start_timestamp;
 
 public:
-  Machine(MachineIDType machine_id, MachineType machine_type, uint8_t control_pin, bool control_pin_active_low);
+  Machine(MachineConfig config);
```

Comment:
if you use initalization list here, like

Machine(MachineConfig _config): config(_config){}

then you can make the class member config cost, that make sure you cant change ID and other stuff, as those are probably supposed to be const.
You could also unpack it if some are const and other not:

Machine(MachineConfig config): id(config.id), type(config.type), //and so on